app/controllers/agent/UTRErrorController.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.agent
+
+import config.featureswitch.{AgentViewer, FeatureSwitching}
+import config.{FrontendAppConfig, ItvcErrorHandler}
+import controllers.agent.predicates.{ConfirmClientController, SelectClientController}
+import controllers.agent.utils.SessionKeys
+import controllers.predicates.agent.AgentAuthenticationPredicate.Redirect
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import uk.gov.hmrc.http.NotFoundException
+import views.html.agent.UTRError
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class UTRErrorController @Inject()(utrError: UTRError,
+                                    val authorisedFunctions: AuthorisedFunctions)
+                                  (implicit mcc: MessagesControllerComponents,
+                                   val appConfig: FrontendAppConfig,
+                                   val itvcErrorHandler: ItvcErrorHandler,
+                                   val ec: ExecutionContext)
+  extends SelectClientController with FeatureSwitching with I18nSupport {
+
+  def show: Action[AnyContent] = Authenticated.async { implicit request =>
+    implicit user =>
+    if (isEnabled(AgentViewer)) {
+      val clientUTR: Option[String] = request.session.get(SessionKeys.clientUTR)
+      clientUTR match {
```

Comment:
I do not think you will need to do this UTR check again as there is already a check in the entreClientUTR page. Should this just a matter of show this page and also when click the button the submit function can redirect the user back to entreClientUTR page.. And there is no session need to be removed as from the beginning we should not save this error UTR in the header. You can confirm with Alex.